Feature/updating indeed scraper (#166)

sammytheindi opened the pull request:
## Description

Merging fixed indeed scraper into master branch. For more information, see #166 
Additional changes include updates to formatting, adding black and prettier configuration files, and performing minor updates to the repo (versioning, updating modules, etc.). 

Biggest point of discussion is whether we want to keep the current method of scraping each individual job page, or scraping the job list with description summaries. The latter (default implementation in this PR), is **much** faster, though will result in less information in the description. It is possible to manually switch back, but do we want to keep it as the default implementation?

## Context of change

- [x] Software (software that runs on the PC)
- [ ] Library (library that runs on the PC)
- [ ] Tool (tool that assists coding development)
- [ ] Other

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Same tests as in #166 

## Checklist:

- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

sammytheindi committed:
Feature/updating indeed scraper (#166)

* - Updated to mobile endpoints and user agents to prevent CAPTCHA
- Updated parsing of indeed scraper
- Fixed tags not being parsed correctly
- Fixed remoteness not being parsed correctly
- Changed to only scrape the first page of each search by default for speed

* - Updated method of loading user agent files
- Updated user agent file of indeed scraper

* - Updated versions in requirements.txt
- Added in black configuration file for formatting
- Added a pre-commit hook so all contributors will have consistent
  formatting on upload
- Updated all python files to conform to black formatter

* Updated Python version

* More black formatting updates

* - Added prettierrc and prettierignore
- Formatted all files other than python

* Updated prettierignore so prettier can search through subdirectories

* Reset formatting to longer line width

* Reverted to previous commit

* Updating again to longer line width after accounting for missing files

* Updated prettierrc and prettierignore files and reran formatting

* Updated version

PaulMcInnis reviewed (changes requested):
lets put back the markdown + demo file but other than this i'm approving it

sammytheindi commented on jobfunnel/resources/user_agent_list_mobile.txt:
```diff
@@ -0,0 +1,16 @@
+Mozilla/5.0 (iPhone; CPU iPhone OS 17_6_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148 Indeed App 225.0
```
We can actually get rid of the first one, only had it in as a reference. I assume that's what you mean?

sammytheindi committed:
- Reverted Markdown changes
- Reverted settings_USA changes
- Updated readme
- Removed extra user-agent from phone user agents list
- Removed extra comments

PaulMcInnis reviewed (changes requested):
wow makes me so happy to see this working again :) 

a few things

1. we should update readme to say that 3.11 is required at least
2. we need to add `user_agent_mobile.txt` to the `MANIFEST.in` file so that it is properly packaged when you install/run using `pip install .` and the `funnel load -s...` per the instructions for new users in the readme.

sammytheindi committed:
Changed readme to refer to python 3.11 instead of 3.8, and added the mobile user agent list to the MANIFEST.in

sammytheindi commented:
Thanks @PaulMcInnis, we should be good to go! 

As I was updating, I was wondering why this project still uses `setup.py`, instead of the newer standard of `pyproject.toml` + `setup.cfg` ([PEP-518](https://peps.python.org/pep-0518/)). 

I will open a new issue on this as I believe we should probably switch while the project is still small, but would love to hear if there are any reasons not to.

PaulMcInnis commented:
> Thanks @PaulMcInnis, we should be good to go!
> 
> As I was updating, I was wondering why this project still uses `setup.py`, instead of the newer standard of `pyproject.toml` + `setup.cfg` ([PEP-518](https://peps.python.org/pep-0518/)).


Hey there,  no reason other than age really - this is an old project and it needs some love
